File: 0252-meeting-rooms/0252-meeting-rooms.js
/**
 * @param {number[][]} intervals
 * @return {boolean}
 */

const overlaps = (x, y) => {
    return !(x[1] <= y[0] || y[1] <= x[0]);
}

const canAttendMeetings = intervals => {
    intervals.sort((a, b) => a[0] - b[0]);
    for (let i = 0; i < intervals.length - 1; i++) {
        if (overlaps(intervals[i], intervals[i + 1])) return false;
    }
    return true;
};

File: 0252-meeting-rooms/0252-meeting-rooms.py
class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        def sortBy(interval):
            return interval[0]
        def overlap(a, b):
            return not (a[1] <= b[0] or b[1] <= a[0])
        intervals.sort(key=sortBy)
        if len(intervals) == 0:
            return True
        interval = intervals[0]
        right = 1
        while right < len(intervals):
            second = intervals[right]
            if overlap(interval, second):
                return False
            else:
                interval = [min(interval[0], second[0]), max(interval[1], second[1])]
            right += 1
        return True
            
            